liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/PackageServerMojo.java

```diff
@@ -138,73 +183,82 @@ private void validateInclude() throws MojoFailureException {
                 String value = includeValues.get(i);
-                includeValues.set(i, value.trim());
+                if (value.trim().length() > 0) {
+                    includeValues.set(i, value.trim());
+                }
             }
         } else {
             includeValues = new ArrayList<String>();
         }
+        return includeValues;
+    }
 
-        // if jar, validate include options, and add runnable
-        if (packageType.equals("jar")) {
-            if (includeValues.contains("usr") || includeValues.contains("wlp")) {
-                throw new MojoFailureException("Package type jar cannot be used with `usr` or `wlp`.");
-            }
+    /**
+     * Sets `packageFileType` based on include and packageType values. Validates the values specified are compatible.
+     * 
+     * @throws MojoFailureException
+     */
+    private void validateIncludeAndPackageType() throws MojoFailureException {
+        ArrayList<String> includeValues = parseInclude();
 
-            if (!includeValues.contains("runnable")) {
-                includeValues.add("runnable");
+        if (includeValues.size() > 1) {
+            if (includeValues.contains("runnable")) {
+                if (includeValues.contains("wlp") || includeValues.contains("usr")) {
+                    throw new MojoFailureException("The `include` parameter value `runnable` is not valid with `usr` or `wlp`.");
+                }
+            } else {
+                throw new MojoFailureException("The `include` parameter value `" + include + "` is not valid. The `include` parameter can be used with values `all`, `usr`, `minify`, `wlp`, `runnable`, `all,runnable`, and `minify,runnable`.");
             }
         }
 
-        if (includeValues.size() > 0) {
-            include = String.join(",", includeValues);
-        }
+        setPackageFileType(includeValues);
     }
 
     /**
-     * Sets `packageFile` based on specified file type, package dir, and package name
-     * Sets default values for unspecified file type, package dir, and package name
+     * Sets `packageFile` and `packageFileType` based on specified/defaulted package type, package dir, and package name.
+     * Validates the include and packageType values before setting the packageFile and packageFileType.
      * 
      * @throws MojoFailureException
      * @throws IOException
      */
-    private void setPackageFilePath() throws MojoFailureException, IOException {
-        String projectFileType = getPackageFileType();
+    private void setPackageFilePath() throws IOException, MojoFailureException {
+        validateIncludeAndPackageType();
+
         String projectBuildDir = getPackageDirectory();
         String projectBuildName = getPackageName();
-        packageFile = new File(projectBuildDir, projectBuildName + projectFileType);
+        packageFile = new File(projectBuildDir, projectBuildName + packageFileType.getFileExtension());
```

Comment:
Actually looking at this, not even sure if we need to distinguish between the `value` and `fileExtension` in the enum. We could just keep one and do `packageFile = new File(projectBuildDir, projectBuildName + "." + packageFileType.getValue());`

Reply:
No need to do additional string concatenation every time when we know it is an extension.

Reply:
My point is you don't need a `value` and `fileExtension` when `fileExtension` is just `value` prepended with `.`. You can add the `.` in at this point.

Reply:
Any my point is it is a known value that can be set ahead of time. Yes it is a simple calculation, but not necessary.